### ISSUE TITLE
Improve ternaries that decaf to IIFEs

### ISSUE THREAD

**Issue by lemonmade** [Shopify/javascript#158]
From CoffeeScript, we get many of these gross IIFEs that are basically just ternary expressions:

```js
(() => {
  if (foo) {
    return bar;
  } else {
    return baz;
  }
 })()
```

This pattern (IIFE, containing only a conditional, optionally with an `else` clause, where each branch of the conditional returns some value) should be transformed into something like `foo ? bar : baz`, assuming that the containing expression can take a ternary (`return`, assignment, etc should all be fine to accept a ternary in place of this expression).

**Comment by justinthec** [comment]
Do these not autoclose on PR merge?

**Comment by GoodForOneFare** [comment]
You need "Fixes #158" in the commit message: https://help.github.com/articles/closing-issues-via-commit-messages/
